src/types/signature.rs

```diff
@@ -290,22 +229,15 @@ impl Signature {
             pub fn input(&self) -> &TypeRow;
             /// Outputs of the abstract signature
             pub fn output(&self) -> &TypeRow;
-            /// Static inputs of the abstract signature
-            pub fn static_input(&self) -> &TypeRow;
         }
     }
 }
 
 impl Display for AbstractSignature {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        let has_inputs = !(self.static_input.is_empty() && self.input.is_empty());
+        let has_inputs = !self.input.is_empty();
         if has_inputs {
```

Comment:
It should be readable now
```suggestion
        if !self.input.is_empty() {
```